e2e/deployer.go

```diff
@@ -0,0 +1,105 @@
+package e2e
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+)
+
+// deployer interface modify the templates on demand and also help
+// to create or delete templates.
+type deployer interface {
+	replaceNamespaceInTemplate(string) (string, error)
+	runAction(string, string)
+	runActionRookDelete(string, string)
+	changeTemplateAndRunAction(string, string)
+}
+
+type deploy struct{}
+
+func (d deploy) replaceNamespaceInTemplate(filePath string) (string, error) {
+	read, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return "", err
+	}
+	return strings.ReplaceAll(string(read), "namespace: default", fmt.Sprintf("namespace: %s", cephCSINamespace)), nil
+}
+
+func (d deploy) runAction(template, action string) {
+	data, err := d.replaceNamespaceInTemplate(template)
+	if err != nil {
+		e2elog.Failf("failed to read content from %s with error %v", template, err)
+	}
+	_, err = framework.RunKubectlInput(cephCSINamespace, data, ns, action, "-f", "-")
+	if err != nil {
+		e2elog.Failf("failed to %s %q with error %v", action, template, err)
+	}
+}
+
+func (d deploy) runActionRookDelete(template, action string) {
+	data, err := d.replaceNamespaceInTemplate(template)
+	if err != nil {
+		e2elog.Failf("failed to read content from %s with error %v", template, err)
+	}
+	_, err = framework.RunKubectlInput(cephCSINamespace, data, "--ignore-not-found=true", ns, action, "-f", "-")
+	if err != nil {
+		e2elog.Failf("failed to %s %q with error %v", action, template, err)
+	}
+}
+
+func (d deploy) changeTemplateAndRunAction(template, action string) {
+	switch template {
+	case rbdDirPath + csiDriverObject, cephfsDirPath + csiDriverObject:
```

Comment:
this makes deploy specific to only RBD and cephfs not a generic one?

Reply:
I was in a double mind here , that said, as mentioned in the PR description actually we can take out this from common interface and use it seperately, but just very limited cases can cover both atm. so kept that way.  

```

The individual deployers are free to override the methods or extend
their own later. Considering, pretty much all the tasks are covered with
few cases in changeTemplateAndRunAction(), it has been kept inside the
common method, but can be split/override from deployers later if needed.

```

Do we expect any other deployment to come in ? if yes, I can prototype to be in `deployer` specific. @Madhu-1

Reply:
Yes this need to be done for vault also https://github.com/ceph/ceph-csi/blob/devel/e2e/deploy-vault.go